server/src/completion.ts

```diff
@@ -148,10 +148,15 @@ export function asCommitCharacters(kind: ScriptElementKind): string[] | undefine
 export function asResolvedCompletionItem(item: TSCompletionItem, details: tsp.CompletionEntryDetails): TSCompletionItem {
     item.detail = asDetail(details);
     item.documentation = asDocumentation(details);
+    item.deprecated = isDeprecated(details.tags);
     Object.assign(item, asCodeActions(details, item.data.file));
     return item;
 }
 
+function isDeprecated(tags?: tsp.JSDocTagInfo[]): boolean | undefined {
+    return tags?.some(tag => tag.name.toLowerCase() === 'deprecated');
+}
```

Comment:
I'm not sure what's the prevalent convention in this code base as I haven't had time to explore it much yet but I would probably prefer tighter typing (`tags` parameter non-optional, return type only `boolean`).

Not a blocker but consider that instead please.

Reply:
I'm using this in a context where the tags are optional on the object, and I like the syntax of making the argument optional.

Instead of `obj.tags ? isDeprecated(obj.tags) : undefined`, I can just do `isDeprecated(obj.tags)`

Reply:
Yeah, maybe at least lets make the return type be just a boolean.

Reply:
It looks like it would actually require an update of vscode libraries though so probably should do that in a separate PR first.